k9mail/src/main/java/com/fsck/k9/ui/messageview/AttachmentController.java

```diff
@@ -123,6 +123,12 @@ private void saveAttachmentTo(File directory) {
             return;
         }
 
+        if (attachment.size > directory.getFreeSpace()) {
```

Comment:
I believe `attachment.size` contains size information from the header. That's not necessarily the actual file size.